### ISSUE TITLE
TextInputInterface.text-input-focused not send when hiding elements

### ISSUE THREAD

**Issue by Murmele** [slint-ui/slint#11079]
### Bug Description

When a textinput gets hidden. Directly (Testcase 1) or indirectly because of the Parent (Testcase 2). `TextInputInterface.text-input-focused` is not triggered.

This is different to https://github.com/slint-ui/slint/pull/11005 because there the element gets destroyed but here it will be only hidden. It was discussed that this might require a logic to send a FocusOut event when the item gets invisible

### text-input-focused-hide.slint

```slint
// Copyright © SixtyFPS GmbH <info@slint.dev>
// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0

import { LineEdit } from "std-widgets.slint";

// If the lineedit gets hidden, the text-input-focused must be false
// to hide also virtual keyboards and so on

export component TestCase inherits Window {

    property <bool> le_visible;

    callback focus_the_line_edit;
    focus_the_line_edit => {
        timer.running = true;
        le.focus();
    }
    callback set_line_edit_visible_and_focused;
    set_line_edit_visible_and_focused => {
        le.visible = true;
        le.focus();
    }

    timer:= Timer {
        running: false;
        interval: 10s;
        triggered => {
            debug("Hide LineEdit");
            le.visible = false;
            self.running = false;
        }
    }

    le := LineEdit {}

    out property<bool> focused: TextInputInterface.text-input-focused;

    changed focused => {
        debug("Focused changed");
        // timer.running = true; // When running in the livepreview this can be used
    }
}

/*
```rust
use slint::{platform::WindowEvent, LogicalPosition};

let instance = TestCase::new().unwrap();
instance.window().dispatch_event(WindowEvent::PointerMoved { position: LogicalPosition::new(0., 0.) }); // Required to trigger traversing through the item tree
assert!(!instance.get_focused());

instance.window().dispatch_event(WindowEvent::PointerMoved { position: LogicalPosition::new(0., 0.) }); // Required to trigger traversing through the item tree
instance.invoke_focus_the_line_edit();
instance.window().dispatch_event(WindowEvent::PointerMoved { position: LogicalPosition::new(0., 0.) }); // Required to trigger traversing through the item tree
assert!(instance.get_focused());

instance.window().dispatch_event(WindowEvent::PointerMoved { position: LogicalPosition::new(0., 0.) }); // Required to trigger traversing through the item tree
slint_testing::mock_elapsed_time(100000); // in ms
instance.window().dispatch_event(WindowEvent::PointerMoved { position: LogicalPosition::new(0., 0.) }); // Required to trigger traversing through the item tree
assert!(!instance.get_focused());

instance.invoke_set_line_edit_visible_and_focused();
instance.window().dispatch_event(WindowEvent::PointerMoved { position: LogicalPosition::new(0., 0.) }); // Required to trigger traversing through the item tree
assert!(instance.get_focused());*/
```

### text-input-focused-parent-hide.slint

```slint
// Copyright © SixtyFPS GmbH <info@slint.dev>
// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0

import { LineEdit } from "std-widgets.slint";

// If the lineedit gets hidden indirectly because a parent gets hidden, the text-input-focused must be false
// to hide also virtual keyboards and so on

export component TestCase inherits Window {

    property <bool> le_visible;

    callback focus_the_line_edit;
    focus_the_line_edit => {
        timer.running = true;
        le.focus();
    }
    callback set_line_edit_visible_and_focused;
    set_line_edit_visible_and_focused => {
        rect.visible = true;
        le.focus();
    }

    timer:= Timer {
        running: false;
        interval: 100ms;
        triggered => {
            debug("Hide LineEdit");
            rect.visible = false;
            self.running = false;
        }
    }

    rect:= Rectangle {
        le:= LineEdit {}
    }

    out property<bool> focused: TextInputInterface.text-input-focused;
}

/*
```rust
let instance = TestCase::new().unwrap();
assert!(!instance.get_focused());

instance.invoke_focus_the_line_edit();
assert!(instance.get_focused());

assert!(instance.get_focused());
slint_testing::mock_elapsed_time(3000); // in ms
assert!(!instance.get_focused());

instance.invoke_set_line_edit_visible_and_focused();
assert!(instance.get_focused());*/
```

### Environment Details

- Slint Version:1.15
- Platform/OS:Linux
- Programming Language:Rust
- Backend/Renderer:
